src/components/ErrorModal.vue

```diff
@@ -0,0 +1,267 @@
+<template>
+  <div
+    class="error-modal"
+    @click="resolve"
+  >
+    <div class="modal-body" @click.stop>
+      <IconClose
+        class="close-modal"
+        @click="resolve"
+      />
+      <template v-if="step === 1">
+        <FailureIcon />
+        <h1>{{ $t('reportBug.Title') }}</h1>
+        <h2>{{ $t('reportBug.SubTitle') }}</h2>
+        <p class="description">
+          {{ $t('reportBug.Description') }}
+        </p>
+        <div
+          class="detailsButton"
+          @click="toggleDetails"
+        >
+          <IconEye />
+          {{ $t('reportBug.ViewDetails') }}
+        </div>
+        <p v-show="details">
+          {{ JSON.stringify(error) }}
+        </p>
+        <h2 class="reportTitle">
+          {{ $t('reportBug.DescriptionTitle') }}
+        </h2>
+        <textarea
+          v-model="report"
+          class="report-text"
+          :placeholder="$t('reportBug.Placeholder')"
+        />
+        <button
+          class="button cancel"
+          @click="resolve"
+        >
+          {{ $t('reportBug.Cancel') }}
+        </button>
+        <button
+          class="button"
+          @click="send"
+        >
+          {{ $t('reportBug.Send') }}
+        </button>
+      </template>
+      <template v-else>
+        <IconOK />
+        <h1 class="successTitle">
+          {{ $t('reportBug.Success.Title') }}
+        </h1>
+        <h2 class="successSubTitle">
+          {{ $t('reportBug.Success.SubTitle') }}
+        </h2>
+        <p class="successDescription">
+          {{ $t('reportBug.Success.Description') }}
+        </p>
+        <button
+          class="button"
+          @click="resolve"
+        >
+          {{ $t('reportBug.Success.Return') }}
+        </button>
+      </template>
+    </div>
+  </div>
+</template>
+
+<script>
+import { detect } from 'detect-browser';
+import FailureIcon from '../assets/APIError.svg?icon-component';
+import IconClose from '../assets/iconCloseRebranded.svg?icon-component';
+import IconEye from '../assets/iconEye.svg?icon-component';
+import IconOK from '../assets/iconOK.svg?icon-component';
+import { backendFetch } from '../utils/backend';
+
+export default {
+  components: {
+    FailureIcon,
+    IconClose,
+    IconEye,
+    IconOK,
+  },
+  props: {
+    resolve: { type: Function, required: true },
+    error: { type: [Error, PromiseRejectionEvent], required: true },
```

Comment:
why do you need the whole PromiseRejectionEvent, won't [`PromiseRejectionEvent.reason`](https://developer.mozilla.org/en-US/docs/Web/API/PromiseRejectionEvent/reason) be enough?

Reply:
maybe it's also just an Object, because I also pass there an whole error and also stack?